src/main.cpp

```diff
@@ -4646,14 +4646,26 @@ bool AskForOutstandingBlocks(uint256 hashStart)
 }
 
 
+void ClearOrphanBlocks() {
+    LOCK(cs_main);
+    for(auto it = mapOrphanBlocks.begin(); it != mapOrphanBlocks.end(); it++) {
+        delete it->second;
+    }
+    mapOrphanBlocks.clear();
+
+    for(auto it = mapOrphanBlocksByPrev.begin(); it != mapOrphanBlocks.end(); it++) {
+        delete it->second;
```

Comment:
I believe that this two maps point to same allocated block. So this call wil essentially free it second time ->SEGV.

Reply:
That's right, remove the second deletion loop and just do `mapOrphanBlocksByPrev.clear();`.